### PR TITLE
add max_allowed_packet variable definition

### PR DESCRIPTION
**Pull request by psinghal20**
See https://github.com/tikv/tikv/issues/3736.

I have made the default value of the `max_allowed_packet` to 4MB as specified by the MySQL specifications [here](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_allowed_packet).

### CODE REVIEW

**Comment by psinghal20** [comment]
@breeswish Can you take a look at this? Thanks!

File: proto/select.proto [hunk 1]
```diff
@@ -104,6 +104,10 @@ message DAGRequest {
 	// Execution summaries will be collected into `execution_summaries` field
 	// in the response.
 	optional bool collect_execution_summaries = 12;
+
+	// It represents the maximum size of one packet or any generated string or any parameter sent as long data.
+	// The default is 4MB in MySQL.
+	optional uint64 max_allowed_packet = 4194304;
```
Comment (breeswish) [inline proto/select.proto]: The value here should be 13 (the label index) instead of 4194304.

File: proto/select.proto [hunk 2]
```diff
@@ -104,6 +104,9 @@ message DAGRequest {
 	// Execution summaries will be collected into `execution_summaries` field
 	// in the response.
 	optional bool collect_execution_summaries = 12;
+
+	// It represents the maximum size of one packet or any generated string or any parameter sent as long data.
```
Comment (Hoverbear) [inline proto/select.proto]:
```suggestion
	// Represents the maximum size of one packet, any generated string, or any parameter sent as long data.
```

**Comment by Hoverbear** [comment]
Rest LGTM

**Comment by breeswish** [comment]
@zz-jason @lonng PTAL. I don't have write permission for this branch.

**Comment by lonng** [comment]
@breeswish I don't have write permission too. 😢

**Pull request merged**